roles/network_plugin/calico/tasks/check.yml

```diff
@@ -0,0 +1,37 @@
+---
+- name: "Check vars defined correctly"
+  assert:
+    that:
+      - "calico_pool_name is defined"
+      - "calico_pool_name | match('^[a-zA-Z0-9-_\\\\.]{2,63}$')"
+      - "ipip_mode is defined"
+      - "ipip_mode in ['Always', 'CrossSubnet', 'Never']"
+    msg: "Check variable definitions seems something is wrong"
+  run_once: yes
+
+- name: "Get current version of calico cluster version"
+  shell: "{{ bin_dir }}/calicoctl version  | grep 'Cluster Version' | awk '{ print $3}'"
+  register: calico_version_on_server
+  run_once: yes
+  delegate_to: "{{ groups['kube-master'][0] }}"
+
+- name: "Determine if calico upgrade is needed"
```

Comment:
I would let the choice to perform or not the migration via kubespray

Reply:
this task cannot fail, it's just output checking and registering variable, if output is empty, then next block does not execute, cause there is ` when:    - 'calico_version_on_server.stdout != ""' `

Reply:
https://github.com/kubernetes-incubator/kubespray/issues/3238 @Atoms

Reply:
oh, when i tested there was no such output :/